src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNullVisitor.java

```diff
@@ -60,9 +60,15 @@ public J visitMethodInvocation(J.MethodInvocation method, P p) {
         if (m.getSelect() != null &&
             !(m.getSelect() instanceof J.Literal) &&
             !m.getArguments().isEmpty() &&
-            m.getArguments().get(0) instanceof J.Literal &&
+            (m.getArguments().get(0) instanceof J.Literal || m.getArguments().get(0) instanceof J.FieldAccess) &&
```

Comment:
This is the relevant change here; in theory that FieldAccess might itself be nullable, whereas the old left hand side is not. That makes this a challenging change to make reliably without potentially making things worse as well.

Reply:
Added a check for the flags of the field type; with that we're good to merge.